src/hooks/index.ts

```diff
@@ -92,23 +96,28 @@ export const useCrashReports = () => {
 
 // WALLET
 export const useWallet = (wallet: YoroiWallet, event: WalletEvent['type']) => {
-  const walletManager = useWalletManager()
   const [_, rerender] = React.useState({})
+  useWalletEvent(wallet, event, () => rerender(() => ({})))
+}
+
+export const useWalletEvent = (wallet: YoroiWallet, event: WalletEvent['type'], callback: () => void) => {
+  const walletManager = useWalletManager()
 
   React.useEffect(() => {
     const unsubWallet = wallet.subscribe((subscriptionEvent) => {
       if (subscriptionEvent.type !== event) return
-      rerender(() => ({}))
+      callback()
     })
     const unsubWalletManager = walletManager.subscribe((subscriptionEvent) => {
       if (subscriptionEvent.type !== event) return
-      rerender(() => ({}))
+      callback()
     })
 
     return () => {
       unsubWallet()
       unsubWalletManager()
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
never do this

Reply:
if you cant find a way, ask for help

Reply:
if you have to, `useCallback` in the calling hook

Reply:
eventually there will be a hook specifically for callbacks (other than useCallback)
but until then, only wrap them when you know its used in a useEffect